src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -534,12 +539,27 @@ export class SavedSearchEmbeddable
         </I18nProvider>,
         domNode
       );
-    }
 
-    this.updateOutput({
-      ...this.getOutput(),
-      rendered: true,
-    });
+      const hasError = this.getOutput().error !== undefined;
+
+      if (
+        this.searchProps!.isLoading === false &&
+        props.searchProps.rows &&
+        props.searchProps.rows.length
```

Comment:
It looks like we used to always update the output to `rendered: true`, but now we have some conditions. Do we know if this will have any impact outside testing? Also, does it make sense that we never indicate a completed render if the selected time range returns no documents (`props.searchProps.rows.length === 0`)?

Reply:
good catch Davis, I also assume this could be reduced to a single

```
this.updateOutput({
           ...this.getOutput(),
           rendered: true,
         });
```

after the conditional `this.renderComplete.dispatchComplete();`  or `this.renderComplete.dispatchError();` is executed

Reply:
@davismcphee @kertal Thanks for the review, I wrote that piece of code and it's only used by the tests in question, so this change is intentional: I was originally looking for a point where I could know that the saved search was rendered, however it seems like I have set the saved search vis as rendered before data actually finished rendering (i.e. when there's still no data and the visualization is not done loading). 

That's why I've moved the `rendered` state into a condition that verifies that rendering is done. 

I'm open to suggestions if you have better ones to determine that the component successfully rendered \ rendered with an error.

Reply:
@lizozom and I synced offline, and she will add a test case for an empty saved search embeddable, too. So it would make sense to simplify 

```
this.updateOutput({
           ...this.getOutput(),
           rendered: true,
         });
```
because it's needed for all conditions